source/_posts/en/tutorial/background.md

```diff
@@ -1,71 +1,93 @@
+JavaScript developer only need to write template including **declare bindings syntax**, the business logic focused on application data in ViewModel. 
 
-- Model: 域模型，用于持久化
-- View: 作为视图模板存在
-- ViewModel: 作为视图的模型，为视图服务
+The template will update automatically when the binding data changed. This design deeply reduces the complexity of the application and increase the efficiency of application development.
 
-### Model 层
+The most significant feature of MVVM pattern is the **data bindings**. It use **data bindings** syntax to separate the View layer from other layers and without any UI code. Unit test code becomes more easily.
 
-Model 层，对应数据层的域模型，它主要做`域模型的同步`。通过 Ajax/fetch 等 API 完成客户端和服务端业务 Model 的同步。在层间关系里，它主要用于抽象出 ViewModel 中视图的 Model。
+After all, MVVM pattern is a Hierarchical architecture, organized as followed: 
 
-### View 层
+- Model: used to store persistence data
+- View: the template including **data bindings** syntax
+- ViewModel: the mirror of template in data ways, auto sync changes with view. 
 
-View 层，作为视图模板存在，在 MVVM 里，整个 View 是一个动态模板。除了定义结构、布局外，它展示的是 ViewModel 层的数据和状态。View 层不负责处理状态，View 层做的是 **数据绑定的声明**、 **指令的声明**、 **事件绑定的声明**。
+### Model
 
-### ViewModel 层
+Modal layer. The model used to store data. Use Ajax/fetch API to keep client and server data sync. It used to abstract the View's Model of ViewModel in layer relations. 
 
-ViewModel 层把 View 需要的层数据暴露，并对 View 层的 **数据绑定声明**、 **指令声明**、 **事件绑定声明** 负责，也就是处理 View 层的具体业务逻辑。ViewModel 底层会做好绑定属性的监听。当 ViewModel 中数据变化，View 层会得到更新；而当 View 中声明了数据的双向绑定（通常是表单元素），框架也会监听 View 层（表单）值的变化。一旦值变化，View 层绑定的 ViewModel 中的数据也会得到自动更新。
 
-### 前端 MVVM 图示
+### View
+
+View: the template. In MVVM pattern, The whole View is a dynamic template. It shows the ViewModel's data and state besides of structure and layout. View does not deal with component state, the main character of View is to declare data bindings, commands, and events.
+
+### ViewModel
+
+ViewModel exports the data which View needed and control View's data bindings, commands, event declaring. 
+
+Developers write custom business logic code to process the data from View. When the data from ViewModel changed, View will get updated automatically; When View declare the two-way bindings syntax, the framework will add the event handler to the DOM, and update the ViewModel's data when DOM get user's input.
+
+### MVVM diagram
 
 <img src="https://raw.githubusercontent.com/X-Jray/blog/master/assets/mvvm.png" width="540" alt="前端MVVM">
 
-如图所示，在前端 MVVM 框架中，往往没有清晰、独立的 Model 层。在实际业务开发中，我们通常按 **Web Component** 规范来组件化的开发应用，Model 层的域模型往往分散在在一个或几个 Component 的 ViewModel 层，而 ViewModel 层也会引入一些 View 层相关的中间状态，目的就是为了更好的为 View 层服务。
+In MVVM framework. There is no distinct boundary of which kinds of code belong to Model. In actual production development, we usually use **Web Component** standard to manage our code. The concept of Model sometimes been separated in one or many component's ViewModel, and ViewModel have some temporary state, to better serve the View layer/
```

Comment:
In MVVM framework. There is no distinct boundary of which kinds of code belong to Model &rarr; There is no clear boundary to determine which kind of code belongs to Model in MVVM frameworks.